### PR TITLE
Change logic to pick which uploads to process

### PR DESCRIPTION
**Pull request by psrikanth88**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: warehouse/warehouse.go [hunk 1]
```diff
@@ -453,13 +462,18 @@ func (wh *HandleT) getPendingUploads(warehouse warehouseutils.WarehouseT) ([]Upl
 	return uploads, nil
 }
 
+func namespaceIdentifier(warehouse warehouseutils.WarehouseT) string {
```
Comment (sivashanmukh) [inline warehouse/warehouse.go:465]: Why not use workerIdentifier itself instead?

Comment (psrikanth88) [reply warehouse/warehouse.go]: @sivashanmukh missed that the func exists, used that

File: warehouse/warehouse.go [hunk 2]
```diff
@@ -578,25 +589,6 @@ func (wh *HandleT) processJobs(warehouse warehouseutils.WarehouseT) (err error)
 		delete(inRecoveryMap, warehouse.Destination.ID)
 	}
 
-	// Step 2: Handle any Pending uploads
-	// An upload job is pending if it is neither exported nor aborted
-
-	pendingUploads, err := wh.getPendingUploads(warehouse)
```
Comment (sivashanmukh) [inline warehouse/warehouse.go:584]: This functions definition can be removed since its not used anywhere else

File: warehouse/warehouse.go [hunk 3]
```diff
@@ -578,25 +589,6 @@ func (wh *HandleT) processJobs(warehouse warehouseutils.WarehouseT) (err error)
...
-	if err != nil {
-		pkgLogger.Errorf("[WH]: Failed to get pending uploads: %s with error %w", warehouse.Identifier, err)
-		return err
-	}
-
-	uploadJob, err := wh.getUploadJobsForPendingUploads(warehouse, whManager, pendingUploads)
```
Comment (sivashanmukh) [inline warehouse/warehouse.go:590]: This functions definition can be removed since its not used anywhere else

File: warehouse/warehouse.go (hunk 2 above)
Comment (psrikanth88) [reply warehouse/warehouse.go:584]: removed

File: warehouse/warehouse.go (hunk 3 above)
Comment (psrikanth88) [reply warehouse/warehouse.go:590]: removed

**Review by sivashanmukh — APPROVED** [review]
LGTM

File: warehouse/warehouse.go [hunk 4]
```diff
@@ -732,15 +641,139 @@ func (wh *HandleT) mainLoop() {
 	}
 }
 
+func (wh *HandleT) getUploadsToProcess(availableWorkers int, skipIdentifiers []string) ([]*UploadJobT, error) {
+
+	var skipIdentifiersSQL string
+	if len(skipIdentifiers) > 0 {
+		skipIdentifiersSQL = fmt.Sprintf(`and concat(t.namespace, ':', t.destination_id) != ALL($1)`)
```
Comment (sivashanmukh) [inline warehouse/warehouse.go:648]: This concat should be with _ instead of column, because that's what we get from workerIdentifier func

**Review by SumanthPuram — APPROVED** [review]

**Pull request merged**